src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -559,11 +612,90 @@ void IgnRenderer::Render()
   }
 }
 
+/////////////////////////////////////////////////
+bool IgnRenderer::GeneratePreviewModel(const std::string &_modelSdfString)
+{
+  sdf::Root root;
+  root.LoadSdfString(_modelSdfString);
+
+  for (auto i = 0u; i < root.ModelCount(); i++)
+  {
+    sdf::Model model = *(root.ModelByIndex(i));
+    if (i == 0)
+      this->dataPtr->previewModelPose = model.Pose();
+    model.SetName(ignition::common::Uuid().String());
+    Entity modelId = this->UniqueId();
+    if (!modelId)
+    {
+      this->TerminatePreviewModel();
+      return false;
+    }
+    this->dataPtr->spawnPreviewModel =
```

Comment:
looks like `spawnPreviewModel` will get overridden and always points to the last model in `root` but the `previewModelPose` points to the pose of the first model. Maybe we can limit the functionality to only load the first model in root?

Reply:
Good catch, updated here a0e82b0